source/common/event/dispatcher_impl.cc

```diff
@@ -121,8 +125,20 @@ Network::DnsResolverSharedPtr DispatcherImpl::createDnsResolver(
     const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers,
     const bool use_tcp_for_dns_lookups) {
   ASSERT(isThreadSafe());
+#ifdef __APPLE__
+  if (Runtime::LoaderSingleton::getExisting()->snapshot().getBoolean(
```

Comment:
I would make this a proper runtime_features flag, and default to on per the other comment.

Reply:
My hesitancy with that was that, it then implies that in the future we will remove the runtime flag and only have this resolver for mac. Not sure if that is exactly what we want given that you can't use tcp or define custom resolver addresses with this?

Reply:
> Not sure if that is exactly what we want given that you can't use tcp or define custom resolver addresses with this?

Hmm. I didn't realize that this resolver doesn't support all options. If that's the case, it really shouldn't be using runtime and it should be a config option that only works on OSX. My personal feeling is that no one will care that these options are not supported on OSX. What I would is actually warn if the below options are set saying they will be ignored, and use runtime feature default to on, and then we can ultimately just make this the default. You should also update the docs for the options in question to indicate they don't work on OSX.

Reply:
ok sounds like a plan. Update coming